packages/front-end/components/License/OptInModal.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useState } from "react";
+import { Box, Text } from "@radix-ui/themes";
+import { AgreementType } from "back-end/src/validators/agreements";
+import { useAuth } from "@/services/auth";
+import usePermissionsUtil from "@/hooks/usePermissionsUtils";
+import Modal from "@/components/Modal";
+import { useUser } from "@/services/UserContext";
+
+// hard coded agreements for now:
+const agreements: Record<
+  AgreementType,
+  {
+    title: string;
+    subtitle: string;
+    terms: React.ReactNode;
+    noPermission: React.ReactNode;
+    version: string;
+  }
+> = {
+  ai: {
+    title: "Enable AI features?",
+    subtitle: "Please read and agree to the terms before proceeding.",
+    terms: (
+      <>
+        This feature involves artificial intelligence technologies, which may
+        include, but are not limited to, sharing information with trusted
+        third-party providers, automated recommendations, content generation, or
+        data analysis.
+        <br />
+        Please select an option below:
+        <br />
+        <ul style={{ marginTop: "10px" }}>
+          <li>
+            I Agree (Opt-In): I consent to the use of artificial intelligence.
+          </li>
+          <li>No thanks (Opt-Out): I do not consent.</li>
+        </ul>
+        For more information about how your data is used, please review our{" "}
+        <a
+          href="https://www.growthbook.io/legal/privacy-policy/06-19-2025"
+          target="_blank"
+          rel="noreferrer"
+        >
+          Privacy Notice
+        </a>{" "}
+        and OpenAI&apos;s{" "}
+        <a
+          href="https://openai.com/enterprise-privacy/"
+          target="_blank"
+          rel="noreferrer"
+        >
+          privacy policy
+        </a>
+        . You can disable these features at any time in your account settings.
+      </>
+    ),
+    noPermission: (
+      <>
+        You must be an administrator to enable this feature. Please contact your
+        administrator.
+      </>
+    ),
+    version: "2025-06-19",
+  },
+  "managed-warehouse": {
+    // STOPSHIP(graham): Update this text to match the actual agreement
```

Comment:
@graham I added this to fix `yarn type-check`.  Though maybe it doesn't belong in this PR at all.